irohad/consensus/yac/storage/yac_storage_cleanup_strategy_impl.hpp

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_YAC_STORAGE_CLEANUP_STRATEGY_IMPL_HPP
+#define IROHA_YAC_STORAGE_CLEANUP_STRATEGY_IMPL_HPP
+
+#include "consensus/yac/storage/yac_storage_cleanup_strategy.hpp"
+
+#include <queue>
+
+#include "consensus/yac/outcome_messages.hpp"
+
+namespace iroha {
+  namespace consensus {
+    namespace yac {
+      class BufferedCleanupStrategy : public CleanupStrategy {
+       public:
+        using QueueSizeType = size_t;
+        using RoundType = Round;
+        using QueueType = std::queue<Round>;
+
+        explicit BufferedCleanupStrategy(QueueSizeType max_limit,
+                                         RoundType minimal_round,
+                                         QueueType existed_state);
+
+        boost::optional<CleanupStrategy::RoundsType> finalize(
+            RoundType round, Answer answer) override;
+
+        bool shouldCreateRound(const Round &round) override;
+
+       private:
+        /// maximal size of queue
+        const QueueSizeType kMaxSize_;
```

Comment:
```suggestion
        const QueueSizeType rounds_amount_limit_;
```

Reply:
rounds amount limit is the maximum size of the queue?

Reply:
Yes, it is.